actors/miner/tests/util.rs

```diff
@@ -1091,6 +1096,279 @@ impl ActorHarness {
+                METHOD_SEND,
+                RawBytes::default(),
+                penalty.clone(),
+                RawBytes::default(),
+                ExitCode::Ok,
+            );
+        }
+
+        let params = ApplyRewardParams { reward: amt, penalty: penalty };
+        rt.call::<Actor>(Method::ApplyRewards as u64, &RawBytes::serialize(params).unwrap())
+            .unwrap();
+        rt.verify();
+    }
+
+    pub fn get_locked_funds(&self, rt: &MockRuntime) -> TokenAmount {
+        let state = self.get_state(rt);
+        state.locked_funds
+    }
+
+    pub fn advance_and_submit_posts(&self, rt: &mut MockRuntime, sectors: &[SectorOnChainInfo]) {
+        // Advance between 0 and 48 deadlines submitting window posts where necessary to keep
+        // sectors proven.  If sectors is empty this is a noop. If sectors is a singleton this
+        // will advance to that sector's proving deadline running deadline crons up to and
+        // including this deadline. If sectors includes a sector assigned to the furthest
+        // away deadline this will process a whole proving period.
+        let state = self.get_state(rt);
+
+        // this has to go into the loop or else go deal with the borrow checker (hint: you lose)
+        //let sector_arr = Sectors::load(&rt.store, &state.sectors).unwrap();
+        let mut deadlines: BTreeMap<u64, Vec<SectorOnChainInfo>> = BTreeMap::new();
+        for sector in sectors {
+            let (dlidx, _) =
+                state.find_sector(&rt.policy, &rt.store, sector.sector_number).unwrap();
+            match deadlines.get_mut(&dlidx) {
+                Some(dl_sectors) => {
+                    dl_sectors.push(sector.clone());
+                }
+                None => {
+                    deadlines.insert(dlidx, vec![sector.clone()]);
+                }
+            }
+        }
+
+        let mut dlinfo = self.current_deadline(rt);
+        while deadlines.len() > 0 {
+            match deadlines.get(&dlinfo.index) {
+                None => {}
+                Some(dl_sectors) => {
+                    let mut sector_nos = BitField::new();
+                    for sector in dl_sectors {
+                        sector_nos.set(sector.sector_number);
+                    }
+
+                    let dl_arr = state.load_deadlines(&rt.store).unwrap();
+                    let dl = dl_arr.load_deadline(&rt.policy, &rt.store, dlinfo.index).unwrap();
+                    let parts = Array::<Partition, _>::load(&dl.partitions, &rt.store).unwrap();
+
+                    let mut partitions: Vec<PoStPartition> = Vec::new();
+                    let mut power_delta = PowerPair::zero();
+                    parts
+                        .for_each(|part_idx, part| {
+                            let sector_arr = Sectors::load(&rt.store, &state.sectors).unwrap();
+                            let live = part.live_sectors();
+                            let to_prove = &live | &sector_nos;
```

Comment:
we want intersection, so I think this is `&` not `|`, at least assuming rust and the bitfield are doing things reasonably

Reply:
ah yes, that a bug.

Reply:
fixed.